tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,336 @@
+ ****************************************************/
+
+/**
+ * @brief Get a handle to operate a "GstInputSelector / GstOutputSelector" node of nnstreamer pipelines
+ * @detail Refer to gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-bad-plugins/html/gst-plugins-bad-plugins-input-selector.html for input selectors
+ *         Refer to https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gstreamer-plugins/html/gstreamer-plugins-output-selector.html for output selectors
+ * @param[in] pipe The pipeline to be managed.
+ * @param[in] switchname The name of switch (InputSelector/OutputSelector)
+ * @param[out] num_nodes The number of nodes selected by this switch node. We assume they are listed from 0 ... num_nodes-1.
+ * @param[in] type The type of the switch.
+ * @param[out] h The switch handle.
+ * @return 0 on success (buf is filled). otherwise a negative error value.
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_switch_gethandle
+(nns_pipeline_h pipe, const char *switchname, int *num_nodes, nns_switch_type type, nns_switch_h *h);
+
+/**
+ * @brief Close the given switch handle
+ * @param[in] h The handle to be closed.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_switch_puthandle (nns_switch_h h);
+
+/**
+ * @brief Control the switch with the given handle to select input/output nodes(pads).
+ * @param[in] h The switch handle returned by nns_pipeline_switch_gethandle()
+ * @param[in] node The node number, which is 0 ... num_nodes-1.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_switch_select (nns_switch_h h, int node);
+
+/**
+ * @brief Get a handle to operate a "GstValve" node of nnstreamer pipelines
+ * @detail Refer to https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gstreamer-plugins/html/gstreamer-plugins-valve.html for more info.
+ * @param[in] pipe The pipeline to be managed.
+ * @param[in] valvename The name of valve (Valve)
+ * @param[out] h The valve handle.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_valve_gethandle
+(nns_pipeline_h pipe, const char *valvename, nns_valve_h *h);
+
+/**
+ * @brief Close the given valve handle
+ * @param[in] h The handle to be closed.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_valve_puthandle (nns_valve_h h);
+
+/**
+ * @brief Control the valve with the given handle
+ * @param[in] h The valve handle returned by nns_pipeline_valve_gethandle()
+ * @param[in] valve_open 0 to close (stop flow). 1 to open (continue flow)
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_valve_control (nns_valve_h h, int valve_open);
+
+/**
+ * @}
```

Comment:
"@}"??? What does it mean by that?

Reply:
It closes line 36, 37. It's the syntax from Tizen C-API definitions.